@here/harp-mapview/lib/FrustumIntersection.ts

```diff
@@ -66,12 +72,28 @@ export class FrustumIntersection {
         this.m_mapTileCuller = new MapTileCuller(m_camera);
     }
 
+    /**
+     * Return camera used for generating frustum.
+     */
+    get camera(): THREE.PerspectiveCamera {
+        return this.m_camera;
+    }
+
+    /**
+     * Return projection used to convert geo coordinates to world coordinates.
+     */
+    get projection(): Projection {
+        return this.m_projection;
+    }
+
     /**
      * Updates the frustum to match the current camera setup.
      */
-    updateFrustum() {
+    updateFrustum(projectionMatrixOveride?: THREE.Matrix4) {
```

Comment:
I'm not sure this is a good idea, shouldn't the m_camera.projectionMatrix always be valid? Now you are creating a frustum which doesn't align with the camera perspective, this could create some problems?

Reply:
Nope. It is required for our purposes, because we need to probe intersection on the extended camera frustum, just to get idea how frustum should look like precisely after taking into account tiles elevations. I have no other idea how we could solve this chicked-egg problem nicer, although tried about 4 approaches already.
I you have some concepts please share...

Reply:
Ahh yes, I had this issue in Olympia, I understand what you mean now. I made a comment below, I think it needs to be mentioned that this is required to avoid the cyclic dependency between near / far plane and visible tile set.

Reply:
Yes perfect catch, this actually for cyclic dependency stuff. I will explain it in solutioning page related to the ticket.